kolibri/plugins/management/assets/src/state/facilityConfigPageActions.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable prefer-arrow-callback */
+const omit = require('lodash/fp/omit');
+const {
+  FacilityResource,
+  FacilityDatasetResource,
+} = require('kolibri').resources;
+const ConditionalPromise = require('kolibri.lib.conditionalPromise');
+const { samePageCheckGenerator } = require('kolibri.coreVue.vuex.actions');
+const preparePage = require('./preparePage');
+const { PageNames, defaultFacilityConfig, notificationTypes } = require('../constants');
+
+// When app is installed, the Facility is assigned id of `1`. Hardcoded here for now.
+const FACILITY_ID = 1;
+
+function resolveOnlyIfOnSamePage(promises, store) {
+  const ident = x => x;
+  return ConditionalPromise.all(promises)
+  .only(samePageCheckGenerator(store), ident, ident)._promise;
+}
```

Comment:
This function is a little confusing to me - mind adding some comments about what it's doing?

Also, referencing `_promise` seems like a smell because typically `_` means internal-use. We may want to make it public, or create a getter?